src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_LockStatement.cs

```diff
@@ -37,6 +40,41 @@ public override BoundNode VisitLockStatement(BoundLockStatement node)
                     argumentType); //need to have a non-null type here for TempHelpers.StoreToTemp.
             }
 
+            if (argumentType.IsWellKnownTypeLock() &&
```

Comment:
>if (argumentType.IsWellKnownTypeLock() [](http://example.com/codeflow?start=12&length=38)

Should this new behavior be tied to `-langversion:preview` (to catch cases where the project is compiled with .NET 9 but not necessarily the latest compiler)?

Reply:
I think we don't officially support such scenario anyway, so this seems unnecessary. But I'm not sure what we should do here.